sklearn/tree/_criterion.pyx

```diff
@@ -283,7 +283,7 @@ cdef class ClassificationCriterion(Criterion):
 
     cdef void init(self, DOUBLE_t* y, SIZE_t y_stride,
                    DOUBLE_t* sample_weight, double weighted_n_samples,
-                   SIZE_t* samples, SIZE_t start, SIZE_t end) nogil:
+                   SIZE_t* samples, SIZE_t start, SIZE_t end) nogil except *:
```

Comment:
I think it wouldn't hurt to have a int return. The current changes produce a looot of overhead :/ especially when the subfunctions are low level ones that get called numerous times...